scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -455,3 +456,63 @@ def test_from_euler_extrinsic_rotation_313():
         [0.91855865, -0.30618622, -0.250000],
         [0.35355339, 0.35355339, 0.8660254]
     ]))
+
+
+def test_as_euler_extrinsic_rotation():
+    np.random.seed(0)
+    n = 10
+    angle1 = np.random.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angle3 = np.random.uniform(low=-np.pi, high=np.pi, size=(n,))
+    # We can enforce different range restrictions on angle2 for different cases
+    angle2_pi = np.random.uniform(low=0, high=np.pi, size=(n,))
+    angle2_pi_2 = np.random.uniform(low=-np.pi / 2, high=np.pi / 2, size=(n,))
+
+    # All axes unique
+    angles_diff_axes = np.column_stack((angle1, angle2_pi_2, angle3))
+    for seq_tuple in permutations('xyz'):
+        seq = ''.join(seq_tuple)
+        assert_allclose(
+            angles_diff_axes,
+            Rotation.from_euler(
+                seq, angles_diff_axes).as_euler(seq)
+        )
+
+    # First and last axes same
+    angles_same_axes = np.column_stack((angle1, angle2_pi, angle3))
+    for se_tuple in permutations('xyz', 2):
+        # First and last axes are same
+        seq = ''.join(se_tuple) + se_tuple[0]
+        assert_allclose(
+            angles_same_axes,
+            Rotation.from_euler(
+                seq, angles_same_axes).as_euler(seq))
+
+
+def test_as_euler_intrinsic_rotation():
+    np.random.seed(0)
+    n = 10
+    angle1 = np.random.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angle3 = np.random.uniform(low=-np.pi, high=np.pi, size=(n,))
+    # We can enforce different range restrictions on angle2 for different cases
+    angle2_pi = np.random.uniform(low=0, high=np.pi, size=(n,))
+    angle2_pi_2 = np.random.uniform(low=-np.pi / 2, high=np.pi / 2, size=(n,))
+
+    # All axes unique
+    angles_diff_axes = np.column_stack((angle1, angle2_pi_2, angle3))
+    for seq_tuple in permutations('XYZ'):
+        seq = ''.join(seq_tuple)
+        assert_allclose(
+            angles_diff_axes,
+            Rotation.from_euler(
+                seq, angles_diff_axes).as_euler(seq)
+        )
+
+    # First and last axes same
+    angles_same_axes = np.column_stack((angle1, angle2_pi, angle3))
+    for se_tuple in permutations('XYZ', 2):
+        # First and last axes are same
+        seq = ''.join(se_tuple) + se_tuple[0]
+        assert_allclose(
+            angles_same_axes,
+            Rotation.from_euler(
+                seq, angles_same_axes).as_euler(seq))
```

Comment:
Besides those tests you definitely should test for degenerate case, where the first and third angles can't be determined uniquely.

Reply:
So I tested dcm equality for the degenerate case. The dcms are not equal upto 6 decimal places. Since it appears that the algorithm is not numerically stable for this case, will it be better to raise a `Warning` / `Error`? Or do we look for a better algorithm?